### ISSUE TITLE
Allow for removal of "pending" source accounts

### ISSUE THREAD

**Issue by zenmonkeykstop** [freedomofpress/securedrop#6488]
## Description

When a source account is created but before a first submission, by default a "pending" flag is set. Pending accounts are not displayed in the journalist interface.

If a user then exits their session and never returns to provide a submission, the account will remain in a pending state indefinitely, and cannot be removed via the Journalist Interface. On long-running high-volume instances this increases the source user count significantly, requiring admin intervention to fix.

One approach to allow for the removal of unused source accounts would be to change the `pending` boolean flag to a datetime value, and purge unused accounts via a cronjob or similar after a set time (say, a month). This has the disadvantage of increasing metadata (source account creation time) about prospective sources, though the field could be set to `null` on submission, meaning that said metadata would not be stored for sources that were actually ever active. This could be mitigated somewhat by giving the datetime a resolution on the order of days or weeks.

Alternatively, the fact that active source sessions without submissions would be nuked when the purge ran could just be accepted. 

Note that this would be moot if/when inverted flow changes land, as accounts would then only be created on first submission. So it might not be worth the effort if it was just a temporary measure.

## To do
1. [x] Add `manage.py` command
2. [ ] Add tests, especially for `--keep-most-recent`'s offset logic
3. [ ] Extract standalone script for testing purposes

## User Research Evidence

long-term observations of instance behavior...

**Comment by cfm** [comment]
Today @nathandyer and I mapped out the following approach for implementing this lever:

1. Add a `securedrop.management.sources` command that:
    1. Retrieves `Source`s with [`pending=True`](https://github.com/freedomofpress/securedrop/blob/d01e743d2853d42c47459163dfae05e4540344f1/securedrop/models.py#L66), except for the (parameterized) *n* most recent—that is, except those with the *n* highest `id`s.
        * This constraint adds a margin for recent pending sources without introducing the timestamp suggested above.
    3. For each `Source` in (1), calls [`EncryptionManager.delete_source_key_pair()`](https://github.com/freedomofpress/securedrop/blob/d01e743d2853d42c47459163dfae05e4540344f1/securedrop/encryption.py#L150).
    4. Deletes all `Source`s in (1).
        * To my eye, `Source` should have a deletion method or hook that automatically calls `EncryptionManager.delete_source_key_pair()` to avoid orphaned keys.  But that's out of scope of this ticket, and a bulk operation will be more efficient anyway.

2. Once we're happy with the implementation of this command via `manage.py`, we can easily derive a standalone version for testing purposes (even just in a gist, to avoid cluttering up the branch).  @nathandyer and I agreed that we'd much rather work backwards from the intended `manage.py` structure than start with a standalone script and then formalize it in `manage.py`.

**Comment by nathandyer** [comment]
@cfm I just pushed the progress I've made [here](https://github.com/freedomofpress/securedrop/compare/d01e743d2853...07a58925058a), if you have a moment would you mind reviewing it and letting me know if I'm on the right track? Thanks!

**Comment by cfm** [comment]
Reviewed directly on <https://github.com/freedomofpress/securedrop/commit/07a58925058acb23d101a8563a800210513ea6f1>, @nathandyer.  :-)

**Comment by nathandyer** [comment]
Perfect, thank you so much for that feedback @cfm! Glad to know I'm on the right track. Will take another swing at this in the morning.

**Comment by nathandyer** [comment]
@cfm I've added some [new changes](https://github.com/freedomofpress/securedrop/tree/6488-pending-sources) which should address the issues you found in the first commit, please feel free to take a look when you have a moment. I will note that I still haven't tested these changes first-hand, I'm getting an error on two different machines when I try to directly call the script (I suspect I have newer packages than I'm supposed to and something got deprecated - will continue to troubleshoot).

**Comment by cfm** [comment]
@nathandyer and I paired on this again today, through which we have a working `manage.py remove-pending-sources --keep-most-recent N` as of eb3159232381b1c331275f56ad103aa534b41587 and a testable version as of e04db45937f3f845af22237fb5913282a5ec6df4.  On Monday we'll pair again on the remaining tasks in the checklist I've added above:

2. Add tests, especially for `--keep-most-recent`'s offset logic
3. Extract standalone script for testing purposes
